File: src/main/java/Lec03/Lec03.java
package Lec03;

import Lec02.Person;

public class Lec03 {
    
    public static void main(String[] args) {
    
    }
    
    public void printAgeIfPerson(Object obj) {
        if (obj instanceof Object) {
            Person person = (Person) obj;
            System.out.println(person.getName());
        }
    }
}

File: src/main/kotlin/Lec03/Lec03.kt
import Lec02.Person

fun main() {
    val number1 = 3L
    val number2: Int = number1.toInt()

    val number3 = 3
    val number4 = 6
    val result = number4 / number3.toDouble()
    println(result)

    val number5: Int? = null
    val number6 = number5?.toDouble() ?: 6
    println(number6)

    val person = Person("dev", 10)
    println("name: ${person.name}")

    val name = "developer"
    println("name: $name")

    val str = """
        ABC
        DEF
        ${name}
    """.trimIndent()
    println(str)

    println(str[4])
}

fun printAgeIfPerson(obj: Any?) {
    val person = obj as? Person
    println(person?.age)
}